types/simulation/rand_util.go

```diff
@@ -147,31 +147,8 @@ func RandSubsetCoins(r *rand.Rand, coins sdk.Coins) sdk.Coins {
 }
 
 // DeriveRand derives a new Rand deterministically from another random source.
-// Unlike rand.New(rand.NewSource(seed)), the result is "more random"
-// depending on the source and state of r.
 //
 // NOTE: not crypto safe.
 func DeriveRand(r *rand.Rand) *rand.Rand {
-	const num = 8 // TODO what's a good number?  Too large is too slow.
-	ms := multiSource(make([]rand.Source, num))
-
-	for i := 0; i < num; i++ {
-		ms[i] = rand.NewSource(r.Int63())
-	}
-
-	return rand.New(ms)
-}
-
-type multiSource []rand.Source
-
-func (ms multiSource) Int63() (r int64) {
-	for _, source := range ms {
-		r ^= source.Int63()
-	}
-
-	return r
-}
-
-func (ms multiSource) Seed(_ int64) {
-	panic("multiSource Seed should not be called")
+	return rand.New(rand.NewSource(r.Int63()))
```

Comment:
I think this function should actually return the same pointer. This allows the fuzzer to have direct control on the bytes produced and try to mutate them in meaningful ways. The code right now will produce a new random source, which will not be directly controlled by the fuzzer, reducing a lot its freedom to explore. This workaround will cause this function to be removed (and tests, since they are no longer needed as well). 

There is a catch, `DeriveRand` is used to give each operation its own random source to be produced:
```golang
for i := 0; i < blocksize; i++ {
        opAndRz = append(opAndRz, opAndR{
                op:   selectOp(r),
                rand: simulation.DeriveRand(r),
        })
}
```
however, using the same pointer will cause an unpredictable number of bytes to be used from each operation. This is not a blocker, but it will produce inputs that are hard to mutate. For instance, if we have two inputs with two transactions each one, where:

 1. op1 consumes 5 bytes and op2 consumes 3 bytes
 2. op1 consumes 2 bytes and op2 consumes 5 bytes

The fuzzer cannot splice the input 1 and input 2 in any meaningful way, making the exploration to be less reliable. We can fix this creating random sources that uses an array of bytes for each operation, but I think it is better to leave this for a future PR, since it requires some experimentation.

Reply:
In summary, I think we should use:
```go
for i := 0; i < blocksize; i++ {
        opAndRz = append(opAndRz, opAndR{
                op:   selectOp(r),
                rand: r,
        })
}
```

and remove `DeriveRand` for now.

Reply:
Forgot to push the change